File: ios/App/App/TokenBridgePlugin.swift
import Capacitor
import WidgetKit

@objc(TokenBridgePlugin)
public class TokenBridgePlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "TokenBridgePlugin"
    public let jsName = "TokenBridge"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "setToken", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "clearToken", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getToken", returnType: CAPPluginReturnPromise),
    ]

    @objc func setToken(_ call: CAPPluginCall) {
        guard let token = call.getString("token") else {
            call.reject("Token is required")
            return
        }
        SharedDataManager.setToken(token)
        WidgetCenter.shared.reloadAllTimelines()
        call.resolve()
    }

    @objc func clearToken(_ call: CAPPluginCall) {
        SharedDataManager.clearToken()
        WidgetCenter.shared.reloadAllTimelines()
        call.resolve()
    }

    @objc func getToken(_ call: CAPPluginCall) {
        let token = SharedDataManager.getToken()
        call.resolve(["token": token as Any])
    }
}

File: android/app/src/main/java/com/lovein/university/TokenBridgePlugin.kt
package com.lovein.university

import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin

@CapacitorPlugin(name = "TokenBridge")
class TokenBridgePlugin : Plugin() {

    companion object {
        const val PREFS_NAME = "lock_widget_prefs"
        const val KEY_JWT_TOKEN = "jwt_token"
    }

    @PluginMethod
    fun setToken(call: PluginCall) {
        val token = call.getString("token")
        if (token == null) {
            call.reject("Token is required")
            return
        }
        val prefs = context.getSharedPreferences(PREFS_NAME, android.content.Context.MODE_PRIVATE)
        prefs.edit().putString(KEY_JWT_TOKEN, token).apply()

        // Trigger immediate widget updates
        LockWidgetProvider.triggerUpdate(context)
        ScheduleWidgetProvider.triggerUpdate(context)
        ManagedLockWidgetProvider.triggerUpdate(context)

        call.resolve()
    }

    @PluginMethod
    fun clearToken(call: PluginCall) {
        val prefs = context.getSharedPreferences(PREFS_NAME, android.content.Context.MODE_PRIVATE)
        prefs.edit().remove(KEY_JWT_TOKEN).apply()

        // Trigger widget updates to show logged-out state
        LockWidgetProvider.triggerUpdate(context)
        ScheduleWidgetProvider.triggerUpdate(context)
        ManagedLockWidgetProvider.triggerUpdate(context)

        call.resolve()
    }

    @PluginMethod
    fun getToken(call: PluginCall) {
        val prefs = context.getSharedPreferences(PREFS_NAME, android.content.Context.MODE_PRIVATE)
        val token = prefs.getString(KEY_JWT_TOKEN, null)
        val ret = JSObject()
        ret.put("token", token ?: JSObject.NULL)
        call.resolve(ret)
    }
}
